### PR TITLE
chore: correct RepeatedField origin

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

TRIVIAL AS IS

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/servers/src/repeated_field.rs [hunk 1]
```diff
@@ -18,8 +18,9 @@
 // OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
 // OR OTHER DEALINGS IN THE SOFTWARE.
 
-// The [Clear] trait and [RepeatedField] are taken from [rust-protobuf](https://github.com/stepancheg/rust-protobuf/tree/master/protobuf-examples/vs-prost)
```
Comment (tisonkun) [inline src/servers/src/repeated_field.rs:21]: master is unstable. Let's try to link to a stable origin.

File: src/servers/src/repeated_field.rs [hunk 2]
```diff
@@ -121,13 +122,13 @@ impl<T> RepeatedField<T> {
 
     /// View data as slice.
     #[inline]
-    pub fn as_slice<'a>(&'a self) -> &'a [T] {
```
Comment (tisonkun) [inline src/servers/src/repeated_field.rs:124]: Run clippy to every code.

**Review by v0y4g3r — APPROVED** [review]
LGTM

File: src/servers/src/repeated_field.rs [hunk 3]
```diff
@@ -414,12 +414,13 @@ impl<'a, T> IntoIterator for &'a mut RepeatedField<T> {
...
 
-impl<'a, T> IntoIterator for RepeatedField<T> {
+impl<T> IntoIterator for RepeatedField<T> {
     type Item = T;
     type IntoIter = vec::IntoIter<T>;
 
-    fn into_iter(self) -> vec::IntoIter<T> {
-        self.into_iter()
+    fn into_iter(mut self) -> vec::IntoIter<T> {
+        self.vec.truncate(self.len);
```
Comment (waynexia) [inline src/servers/src/repeated_field.rs:422]: What is this for, Is there any lint or test that reminds you of it?

**Review by waynexia — APPROVED** [review]
Others LGTM

File: src/servers/src/repeated_field.rs (hunk 3 above)
Comment (tisonkun) [reply src/servers/src/repeated_field.rs:422]: Clippy complains there are two "into_iter" that can be confusing. Retain the one here.

**Pull request merged**